src/libAtomVM/bif.c

```diff
@@ -200,6 +200,21 @@ term bif_erlang_is_tuple_1(Context *ctx, uint32_t fail_label, term arg1)
     return term_is_tuple(arg1) ? TRUE_ATOM : FALSE_ATOM;
 }
 
+term bif_erlang_is_record(Context *ctx, uint32_t fail_label, term arg1, term arg2)
+{
+    if (!term_is_tuple(arg1) || !term_is_atom(arg2) || term_get_tuple_arity(arg1) < 1) {
+        return FALSE_ATOM;
```

Comment:
When not used as a guard I think we should raise here: as far as I know, if something different than an atom is used for arg2, an error is raised.
Look at `RAISE_ERROR_BIF` or even to `VALIDATE_VALUE_BIF` that validates argument type.